sys/kern/sys_process.c

```diff
@@ -663,9 +663,15 @@ sys_ptrace(struct thread *td, struct ptrace_args *uap)
 		syscallarg_t args[nitems(td->td_sa.args)];
 		struct ptrace_sc_ret psr;
 		int ptevents;
-	} r;
-	void * __capability addr;
-	int error = 0;
+	} r = { 0 };
+
+	union {
+		struct ptrace_lwpinfo_c pl;
+	} c = { 0 };
```

Comment:
Why this union?  Also, we shouldn't be using `_c` types here.

Reply:
A copypaste